### PR TITLE
Journey action steps

### PR DESCRIPTION
**Pull request by cfarm**
## Added
- JS for expandables on process pages
- JS module for process pages
- Action steps data
- Action steps patterns markup, CSS
## Changes
- Layout markup for the expandables content
- Data for the action steps content for Clarify your goals
- Gruntfile legibility
- Checked content styling and alert styling refactored to be used with lists (alert from loan options, checked class from the Phase Milestones at the top of the Journey pages)
## Review
- @virginiacc 
- @huetingj 
- @stephanieosan 
## Todo in future PR's
- bottom-of-expandable close button
### Small screen un-expanded

screen shot 2015-03-20 at 4 14 24 pm
### Small screen expanded step

screen shot 2015-03-20 at 4 13 30 pm
### Medium un-expanded

screen shot 2015-03-20 at 4 16 23 pm
### Medium expanded step

screen shot 2015-03-20 at 4 16 28 pm
### Large un-expanded

screen shot 2015-03-20 at 4 16 43 pm
### Large expanded step

screen shot 2015-03-20 at 4 16 48 pm

### CODE REVIEW

**Comment by stephanieosan** [comment]
Yay!!! These are looking great! 

**Georgia for bullets**
This is a detail thing, but all the bullet-type content (tips, checks, alerts) are designed in georgia (not Avenir) to give them some extra differentiation. They could also use a little bit of space between the bullets. 
screen shot 2015-03-20 at 4 44 50 pm

**Padding feedback**
Maybe this isn't built into Capital Framework yet (and it's totally a detail thing, too), but there should be 60px of space between the intro section (intro paragraph + key tool) and the "How to take action" section. Then another 60px between the "How to take action" and "What to know" sections. Then 60 after the last bit of content and the next action step expandable.

**Comment by mthibos** [comment]
FWIW, this looked weird to my eye:

image

**Comment by mthibos** [comment]
Also, you rock @cfarm!  Does this mean the CMS is all working and jazz?  :dancer: :saxophone: :guitar:

**Comment by stephanieosan** [comment]
Yeah, I noticed what @mthibos brought up too. Is that how the Flapjack pattern works by default? 

What we saw in the design files from that flapjack pattern was this: GHE/OAH/OAH-notes/issues/867#issuecomment-39939 where they dropped the word "Show" and just included the plus icon.

**Comment by cfarm** [comment]
I think they are removing the word "Show" manually based on the line length for the button. We'd want to parse the character length to do this in Sheer - so I'll give that a go.

Just for the record, yes these styles and the markup are directly from Capital Framework - our only alteration was to make the content flush with the container edges instead of indented like the [CF ones](https://cfpb.github.io/cf-expandables/docs/):

screen shot 2015-03-20 at 5 47 28 pm

**Comment by cfarm** [comment]
Now that I look at the designs again I don't think this flush-sides pattern is necessary so going to remove it.

**Comment by cfarm** [comment]
@mthibos CMS isn't quite done yet - we need to get the content from WordPress to Sheer next week, then we can have this actually be dynamic. We're still using placeholder content for now.

**Comment by cfarm** [comment]
Updated the expandable:

1. Heading for Clarify your goals shows number in front - this would be added via CMS
2. Expandable padding matches Capital Framework default
3. Expandable text for button now has a width, removing the extra whitespace below for long text:
screen shot 2015-03-20 at 6 25 16 pm

**Comment by cfarm** [comment]
List styles updated to have Georgie serif font and whitespace between:

screen shot 2015-03-20 at 6 36 24 pm

**Comment by cfarm** [comment]
Andddd padding updates for large screens:

screen shot 2015-03-20 at 7 16 46 pm

**Comment by huetingj** [comment]
The updated small-width un-expanded design looks great! And looking good overall. Somebody tell me that the alert icon spacing looks ok... I know the minicon is the correct size and spacing. My eye has a little trouble with it because it's larger than the green check mark and bullet point. Does it bother you @stephanieosan?

**Comment by huetingj** [comment]
The updated small-width un-expanded design looks great! And looking good overall. I know the alert minicon is the correct size and spacing. But my eye has a little trouble with it because it's larger than the green check mark and bullet point. What do you think @stephanieosan?

**Comment by stephanieosan** [comment]
Yeah, I actually wrestled with this in the design originally, but figured it was okay. 

The only other option I see is indenting the georgia bullet paragraphs even further, so there's more space between all bullets, but that could be visually weird, too. I'm okay with leaving it as-is.

File: src/process/index.html [hunk 1]
```diff
@@ -25,14 +25,29 @@
...
+        <p>If you’re already behind on your payments, contact your servicer right away.</p>
+        <ul class='list__branded list__flush-sides'>
+          <li class='list_item__spaced'>
+            Ask your servicer what options are available to help you get back on track or make it through temporary financial difficulties.
+          </li>
+          <li class='checked list_item__spaced'>
+            Ask your servicer to send you a written explanation of options that may be available to you if you can’t make your payments in the future.    
+          </li>
+          <li class='alert list_item__spaced'>
+            If you are having problems with your servicer, you can submit a complaint with the CFPB.
```
Comment (virginiacc) [inline src/process/index.html]: If these patterns are consistent across all the steps, I think we can probably apply most of the styling automatically so users only have to add the `.alert` & `.checked` classes in WordPress.

File: src/static/css/module/helpers.less [hunk 2]
```diff
@@ -253,11 +253,47 @@ object {
...
+    top: 3px;
+  }
+  p&:before {
+    text-indent: -1.4em;
+  }
+  li& {
+    position: relative;
+    list-style: none;
+  }
 }
```
Comment (virginiacc) [inline src/static/css/module/helpers.less]: I'm wondering if we could create a class (like `.icon-bullet`) that would pull out shared code here (there's also a `.custom-bullet` macro in `.cf-typography`, but that wouldn't really reduce the code.) 

I also notice that I hard-coded the font sizes here; I will look at using relative sizes instead. ;)

File: src/static/css/module/process.less [hunk 3]
```diff
@@ -87,10 +87,13 @@
...
 .expandable__padded.expandable__flush-sides {
+  margin-right: 0;
   .expandable_header {
     padding: unit(10px / @base-font-size-px, em) 0;
   }
-  margin-right: 0;
+  .expandable_content {
+    margin: 0;    
+  }
 }
```
Comment (virginiacc) [inline src/static/css/module/process.less]: I think we can delete this if it's not being used anymore.

**Comment by virginiacc** [comment]
This looks great! I had a few suggestions on how the styles can be applied to the steps, but I think we can pull this in and make changes in a future PR when we start pulling in content from the CMS.

File: src/process/index.html (hunk 1 above)
Comment (cfarm) [inline src/process/index.html]: do you mean the list_item__spaced styling?

**Pull request merged**